### PR TITLE
Feat airports

### PR DESCRIPTION
**Pull request by SteveP2000**

### CODE REVIEW

**Comment by SteveP2000** [comment]
Snakemake not implemented

**Comment by leonbaruah** [comment]
Merging, but needs to be put into a callable function before it can be implemented properly with snakemake etc.

File: src/extract/ourairports.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+airport_df = pd.DataFrame(airport_list)
+print(airport_df)
+airport_df = airport_df.drop(columns=['keywords'])
+
+airport_pt = gpd.GeoDataFrame(airport_df,
+    geometry=gpd.points_from_xy(airport_df.lon, airport_df.lat))
+
+# Set CRS
+airport_pt.crs = {'init': 'epsg:4326'}
```
Comment (leonbaruah) [inline src/extract/ourairports.py:30]: The projection is declared as 4326 as a string - that's fine but is this always true for ourairports?

File: src/extract/wfp.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import requests
+import geopandas as gpd
+import pandas as pd
+
+next_link = "https://geonode.wfp.org/geoserver/wfs?srsName=EPSG%3A4326&typename=geonode%3Awld_trs_airports_wfp&outputFormat=json&service=WFS&request=GetFeature"
+iso3 = 'YEM'
```
Comment (leonbaruah) [inline src/extract/wfp.py:6]: When put into a function, have this as an input

File: src/extract/wfp.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+airport_df = pd.DataFrame(airport_list)
+
+apt_yem = airport_df[airport_df['iso3'] == iso3]
+
+airport_pt = gpd.GeoDataFrame(apt_yem,
+    geometry=gpd.points_from_xy(apt_yem.longitude, apt_yem.latitude))
+
+# Set CRS
+airport_pt.crs = {'init': 'epsg:4326'}
```
Comment (leonbaruah) [inline src/extract/wfp.py:31]: The projection is declared as 4326 as a string - that's fine but is this always true for geonode api?

File: src/extract/wfp.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import requests
+import geopandas as gpd
+import pandas as pd
+
+next_link = "https://geonode.wfp.org/geoserver/wfs?srsName=EPSG%3A4326&typename=geonode%3Awld_trs_airports_wfp&outputFormat=json&service=WFS&request=GetFeature"
```
Comment (leonbaruah) [inline src/extract/wfp.py:5]: This should be in the snakemake file to be called by this function

File: src/extract/ourairports.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+# Export to file
+airport_pt.to_csv("../../raw_data/ourairports_airports.csv", index=False)
+# airport_pt.to_file("../../raw_data/wfp_airports.gpkg", driver="GPKG")
+airport_pt.to_file("../../raw_data/ourairports_airports.shp")
+airport_pt.to_file("../../processed_data/yem_tran_air_pt_s1_ourairports_pp.shp")
```
Comment (leonbaruah) [inline src/extract/ourairports.py:36]: Use the yaml file to declare the directory (and processed output)  to output the files to.

File: src/extract/wfp.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    geometry=gpd.points_from_xy(apt_yem.longitude, apt_yem.latitude))
+
+# Set CRS
+airport_pt.crs = {'init': 'epsg:4326'}
+
+# Export to file
+# airport_pt.to_file("../../raw_data/wfp_airports.gpkg", driver="GPKG")
+airport_pt.to_file("../../raw_data/wfp_airports.shp", encoding='utf-8')
+airport_pt.to_file("../../processed_data/yem_tran_air_pt_s1_wfp_pp.shp", encoding='utf-8')
+airport_pt.to_csv("../../raw_data/wfp_airports.csv", index=False, encoding='utf-8')
```
Comment (leonbaruah) [inline src/extract/wfp.py:37]: Use the yaml file to declare the directory (and processed output) to output the files to.

**Review by leonbaruah — APPROVED** [review]

**Pull request merged**